arviz/plots/dotplot.py

```diff
@@ -0,0 +1,229 @@
+    quartiles=True,
+    point_interval=None,
+    ax=None,
+    show=None,
+    plot_kwargs=None,
+    backend=None,
+    backend_kwargs=None,
+    **kwargs
+):
+    """Plot distribution as dot plot or quantile dot plot.
+
+    This function uses the Wilkinson's Algorithm[1] to allot dots to bins.
+    The quantile dot plots was inspired from the paper[2].
+
+    Parameters
+    ----------
+    values : array-like
+        Values to plot
+    binwidth : float, optional
+        Width of the bin for drawing the dot plot.
+    dotsize : float, optional
+        The size of the dots relative to the bin width. The default, 1, makes dots be
+        just about as wide as the bin width.
+    stackratio : float, optional
+        The distance between the center of the dots in the same stack relative to the bin height.
+        The default, 1, makes dots in the same stack just touch each other.
+    point_interval : bool, optional
+        Plots the point interval. Uses hdi_prob to plot the HDI interval
+    point_estimate : str, optional
+        Plot point estimate per variable. Values should be ‘mean’, ‘median’, ‘mode’ or None.
+        Defaults to ‘auto’ i.e. it falls back to default set in rcParams.
+    dotcolor : string, optional
+        The color of the dots. Should be a valid matplotlib color.
+    intervalcolor : string, optional
+        The color of the interval. Should be a valid matplotlib color.
+    linewidth : int, optional
+        Line width throughout. If None it will be autoscaled based on figsize.
+    markersize : int, optional
+        Markersize throughout. If None it will be autoscaled based on figsize.
+    markercolor: string, optional
+        The color of the marker when plot_interval is True. Should be a valid matplotlib color.
+    marker: string, optional
+        The shape of the marker. Valid for matplotlib backend
+        Defaults to "o"
+    hdi_prob : float, optional
+        Valid only when point_interval is True. Plots HDI for chosen percentage of density.
+        Defaults to 0.94.
+    rotated : bool, optional
+        Whether to rotate the dot plot by 90 degrees.
+    nquantiles : int, optional
+        Number of quantiles to plot, used for quantile dot plots
+        Defaults to 50.
+    quartiles : bool, optional
+        If True then the quartile interval will be plotted with the HDI.
+        Defaults to True.
+    figsize : tuple, optional
+        Figure size. If None it will be defined automatically.
+    plot_kwargs : dict, optional
+        Keywords passed for customizing the dots.
+    backend: str, optional
+        Select plotting backend {"matplotlib","bokeh"}. Default "matplotlib".
+    ax : axes, optional
+        Matplotlib axes or bokeh figures.
+    show: bool, optional
+        Call backend show function.
+    backend_kwargs: bool, optional
+        These are kwargs specific to the backend being used. For additional documentation
+        check the plotting method of the backend.
```

Comment:
Same here, add where these kwargs are passed to. Ideally users won't need to read the source of this function to see where these are passed, then go to plot_utils to the right function and then go to the matplotlib/bokeh docs of the function/class used only to know what is valid here, the same docstring should have the links to matplotlib/bokeh docs directly to the right function. We can then comment on #1188 to have this docstring be an example of good practices

side note, it says bool, I think it should be a dictionary.